### PR TITLE
Log a SoftException on SurfaceMountingManager.addRootView

### PR DESCRIPTION
**Pull request by psionic12**
## Summary
Follow the same solution (do not throw a crash when view ID is set already) used in `ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java` for `ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.java`

## Changelog

[Android] [Changed] - Log a SoftException on SurfaceMountingManager.addRootView

## Test Plan

None

### CODE REVIEW

File: ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.java [hunk 1]
```diff
@@ -231,10 +231,19 @@ public void run() {
...
+              // Hence this can cause unnecessary crashes at runtime for hybrid apps.
+              // So converting this to a soft exception such that pure react-native devs can still see the
+              // warning while hybrid apps continue to run without crashes
+              ReactSoftExceptionLogger.logSoftException(
+                TAG,
+                new IllegalViewOperationException(
+                  "Trying to attach a ReactRootView with an explicit id already set to ["
+                    + rootView.getId()
+                    + "]. React Native uses the id field to track react tags and will overwrite this"
+                    + " field. If that is fine, explicitly overwrite the id field to View.NO_ID."));
```
Comment (cortinico) [inline ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.java:246]: Can we retain the original message?

**Comment by cortinico** [comment]
> 📋 Verify Changelog Format - A changelog entry has the following format: [CATEGORY] [TYPE] - Message.

Can you also specify the correct changelog entry?

**Comment by psionic12** [comment]
@cortinico Done.

**Comment by cortinico** [comment]
Sorry this took so long :(

**Pull request closed without merging**